instrumentation/net/http/httptrace/otelhttptrace/clienttrace_test.go

```diff
@@ -32,3 +34,45 @@ func ExampleNewClientTrace() {
 
 	fmt.Println(resp.Status)
 }
+
+type zeroTripper struct{}
+
+func (zeroTripper) RoundTrip(_ *http.Request) (*http.Response, error) {
+	return &http.Response{StatusCode: 200}, nil
+}
+
+var _ http.RoundTripper = zeroTripper{}
+
+// TestNewClientParallelismWithoutSubspans tests running many Gets on a client simultaneously,
+// which would trigger a race condition if root were not protected by a mutex.
+func TestNewClientParallelismWithoutSubspans(t *testing.T) {
+	t.Parallel()
+
+	makeClientTrace := func(ctx context.Context) *httptrace.ClientTrace {
+		return NewClientTrace(ctx, WithoutSubSpans())
+	}
+
+	client := http.Client{
+		Transport: otelhttp.NewTransport(
+			zeroTripper{},
+			otelhttp.WithClientTrace(makeClientTrace),
+		),
+	}
+
+	var wg sync.WaitGroup
+
+	for i := 1; i < 10000; i++ {
```

Comment:
This is a lot of goroutines being spawned here. What's the reasoning behind this? It is making this test quite resource intensive.